utils/envvars/parser.go

```diff
@@ -163,22 +161,22 @@ func parseStringDuration(value string) (time.Duration, error) {
 
 // GetSAASHost returns SaaS host env variable value if it's present and valid.
 // Otherwise returns defaultSaaSHost.
-func GetSAASHost(fallbackEnv string) (string, error) {
+func GetSAASHost() (string, error) {
 	name, v := envSaaSHost, os.Getenv(envSaaSHost)
 	if v == "" {
-		name, v = fallbackEnv, os.Getenv(fallbackEnv)
-	}
-	if v == "" {
-		logrus.Debugf("Using default SaaS host %q.", defaultSaaSHost)
+		logrus.Infof("Using default SaaS host %q.", defaultSaaSHost)
 		return defaultSaaSHost, nil
 	}
+	if strings.HasPrefix(v, ":") {
+		return "", fmt.Errorf("environment variable %q has invalid format %q. Expected host:[port]", name, v)
+	}
 
 	host, port, err := net.SplitHostPort(v)
-	if err != nil {
+	if err != nil && strings.Count(v, ":") > 1 {
```

Comment:
And that? I think net.SplitHostPort should do everything we need

Reply:
it actually does not. If we pass only hostname without `:[port]`, net.SplitHostPort will fail with empty `host`, `port` and an error. So the code below will not work. Current code will add `443` as port

Reply:
Then I think there should be check for `>= 1`, not `> 1`

Reply:
done